server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/LocalStorageManager.scala

```diff
@@ -663,9 +666,21 @@ private[worker] final class LocalStorageManager(
   }
 
   private def flushFileWriters(): Unit = {
-    writers.entrySet().asScala.foreach(u =>
-      u.getValue.asScala
-        .foreach(f => f._2.flushOnMemoryPressure()))
+    fileMetas.keys().asScala.foreach { shuffleKey =>
+      partitionLocationInfo.getAllMasterLocations(shuffleKey).asScala.foreach { partition =>
```

Comment:
PartitionLocationInfo getAllMasterLocations is a synchronized call, trim actions will be called frequently. I think LocalStorageManager's workingDirWriters contains the information you need.